pkg/sensors/exec/cgroups_test.go

```diff
@@ -67,3 +129,99 @@ func TestTgRuntimeConf(t *testing.T) {
 	assert.Equal(t, ret.TgCgrpSubsysIdx, cgroups.GetCgrpSubsystemIdx())
 	assert.Equal(t, ret.LogLevel, uint32(logger.GetLogLevel()))
 }
+
+// Test we do not receive any cgroup events from BPF side
+func TestCgroupNoEvents(t *testing.T) {
+	testutils.CaptureLog(t, logger.GetLogger().(*logrus.Logger))
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	option.Config.HubbleLib = tus.Conf().TetragonLib
+	option.Config.Verbosity = 5
+
+	_, err := observer.GetDefaultObserver(t, ctx, tus.Conf().TetragonLib)
+	if err != nil {
+		t.Fatalf("GetDefaultObserver error: %s", err)
+	}
+
+	loadedSensors := []*sensors.Sensor{
+		testsensor.GetTestSensor(),
+		testsensor.GetCgroupSensor(),
+	}
+
+	registerSensors(loadedSensors)
+
+	mapDir := bpf.MapPrefixPath()
+	smanager, err := sensors.StartSensorManager(mapDir, mapDir, "")
+	if err != nil {
+		t.Fatalf("startSensorController failed: %s", err)
+	}
+	observer.SensorManager = smanager
+	defer func() {
+		err := smanager.StopSensorManager(ctx)
+		if err != nil {
+			fmt.Printf("stopSensorController failed: %s\n", err)
+		}
+	}()
+
+	err = enableSensors(ctx, smanager, loadedSensors)
+	if err != nil {
+		t.Fatalf("enableSensors error: %s", err)
+	}
+
+	defer func() {
+		err := disableSensors(ctx, smanager, loadedSensors)
+		if err != nil {
+			fmt.Printf("disableSensor failed: %s\n", err)
+		}
+	}()
+
+	val, err := testutils.GetTgRuntimeConf()
+	assert.NoError(t, err)
+
+	// Set Log Level to trace so we receive BPF events
+	val.LogLevel = uint32(logrus.TraceLevel)
```

Comment:
If set set the trace level, why are not we expected to receive bpf events? Is it because they are happening in a level that we are not monitoring?

Reply:
So yes we may not receive events if they happen at level that we do not care about, but for this case we do not expect cgroup events cause we do not probe tetragon to migrate itself and finish the cgroup discovery.

We only detect environment and half of the cgroup setup, but not all of it.

See below
`val.TgCgrpLevel = 3`

Reply:
Improved this, tests are more readable now!